src/vulkan/vk-shader-table.cpp

```
@@ -168,17 +168,35 @@ ShaderTableImpl::PipelineData* ShaderTableImpl::getPipelineData(RayTracingPipeli
+    const uint32_t tableOffset = (uint32_t)(alignedTableAddress - bufferAddress);
+    SLANG_RHI_ASSERT((bufferAddress + tableOffset) % tableAlignment == 0);
+
+    BufferImpl* bufferImpl = checked_cast<BufferImpl*>(buffer.get());
+    if (SLANG_FAILED(device->uploadBufferInitData(bufferImpl, tableOffset, tableSize, tableData.get())))
+    {
+        SLANG_RHI_ASSERT_FAILURE("Failed to upload shader table data");
+        return nullptr;
+    }
+
     RefPtr<PipelineData> pipelineData = new PipelineData();
 
```

Cppcheck (2.21.0)

[style] The function 'capacity' is never used.

(unusedFunction)